api/diagnostics.go

```diff
@@ -519,7 +519,16 @@ func (j *DiagnosticsJob) getHTTPAddToZip(node Node, endpoints map[string]string,
 		}
 
 		if resp.StatusCode != http.StatusOK {
-			errMsg := fmt.Sprintf("unable to fetch %s. Return code %d", fullURL, resp.StatusCode)
+			body, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				logrus.WithError(err).Warn("Unable to read response body")
```

Comment:
Don't you have to close the response body even when `ReadAll` fails? As I understand it, you need to close it whenever you work with the body or _attempt to_, see: https://golang.org/pkg/net/http/

![screenshot 2019-02-14 at 09 44 23](https://user-images.githubusercontent.com/1185253/52774267-21388000-303d-11e9-8424-087c45020837.png)



How about something like this?

```go
var errMsg string
body, err := ioutil.ReadAll(resp.Body)
resp.Body.Close()
if err != nil {
	errMsg = fmt.Sprintf("unable to fetch %s. Return code %d. Could not read body", fullURL, resp.StatusCode)
} else {
	errMsg = fmt.Sprintf("unable to fetch %s. Return code %d. Body: %s", fullURL, resp.StatusCode, string(body))
}

```

Reply:
Thanks